tests/pipelines/test_pipelines_automatic_speech_recognition.py

```diff
@@ -1007,6 +1010,119 @@ def test_whisper_language(self):
             {"text": " Mr. Quilter is the apostle of the middle classes and we are glad to welcome his gospel."},
         )
 
+    @slow
+    def test_speculative_decoding_whisper_non_distil(self):
+        # Load data:
+        dataset = load_dataset("hf-internal-testing/librispeech_asr_dummy", "clean", split="validation[:1]")
+        sample = dataset[0]
+
+        # Load model:
+        model_id = "openai/whisper-large-v2"
+        processor = AutoProcessor.from_pretrained(model_id)
+        model = AutoModelForSpeechSeq2Seq.from_pretrained(
+            model_id,
+            low_cpu_mem_usage=True,
+            use_safetensors=True,
+        )
+
+        # Load assistant:
+        assistant_model_id = "openai/whisper-tiny"
+        assistant_model = AutoModelForSpeechSeq2Seq.from_pretrained(
+            assistant_model_id,
+            low_cpu_mem_usage=True,
+            use_safetensors=True,
+        )
+
+        # Load pipeline:
+        pipe = AutomaticSpeechRecognitionPipeline(
+            model=model,
+            tokenizer=processor.tokenizer,
+            feature_extractor=processor.feature_extractor,
+            generate_kwargs={"language": "en"},
+        )
+
+        inputs = {
+            "sampling_rate": sample["audio"]["sampling_rate"],
+            "raw": np.array(sample["audio"]["array"]),
+        }
+
+        start_time = time.time()
+        transcription_non_ass = pipe(inputs=inputs, generate_kwargs={"assistant_model": assistant_model})["text"]
+        total_time_assist = time.time() - start_time
+
+        inputs = {
+            "sampling_rate": sample["audio"]["sampling_rate"],
+            "raw": np.array(sample["audio"]["array"]),
+        }
+
+        start_time = time.time()
+        transcription_ass = pipe(inputs=inputs)["text"]
+        total_time_non_assist = time.time() - start_time
+
+        assert transcription_ass == transcription_non_ass
+        assert (
+            transcription_ass
+            == " Mr. Quilter is the apostle of the middle classes and we are glad to welcome his gospel."
+        )
+
+        assert total_time_non_assist > total_time_assist, "Make sure that assistant decoding is faster"
+
+    @slow
+    def test_speculative_decoding_whisper_distil(self):
```

Comment:
Feel free to propagate the above changes here as well

Reply:
Done!